### PR TITLE
QUIC: Records STREAM frame

### PR DESCRIPTION
**Pull request by scw00**
QUIC: Records STREAM frame, RST frame, STOP_SENDING frame and MAX_STREAM_DATA frame

### CODE REVIEW

File: iocore/net/quic/QUICStream.h [hunk 1]
```diff
@@ -184,6 +207,16 @@ class QUICCryptoStream : public QUICFrameGenerator
...
 private:
+  struct CryptoFrameInformation {
+    QUICOffset offset;
+    Ptr<IOBufferBlock> block;
+  };
+
+  void _on_frame_acked(QUICFrameInformation info) override;
+  void _on_frame_lost(QUICFrameInformation info) override;
+
+  QUICFrameId _bind_and_record_frame();
```
Comment (maskit) [inline iocore/net/quic/QUICStream.h]: Not used.

File: iocore/net/quic/QUICStream.h [hunk 2]
```diff
@@ -107,6 +126,10 @@ class QUICStream : public VConnection, public QUICFrameGenerator, public QUICTra
   Event *_send_tracked_event(Event *, int, VIO *);
 
   void _write_to_read_vio(QUICOffset offset, const uint8_t *data, uint64_t data_length, bool fin);
+  void _records_max_stream_data_frame(QUICMaxStreamDataFrame *frame);
```
Comment (maskit) [inline iocore/net/quic/QUICStream.h]: Can we make these to receive const references (`const QUICAbcFrame &frame`) ?

**Comment by maskit** [comment]
I assume this PR just records STREAM frame as the first step and the next step (PR) will be retransmitting STREAM frames based on the recorded information. Is it correct?

**Comment by scw00** [comment]
Yep

File: iocore/net/quic/QUICStream.h (hunk 1 above)
Comment (scw00) [reply iocore/net/quic/QUICStream.h]: forgot it.

**Comment by scw00** [comment]
updated!

**Comment by maskit** [comment]
[approve ci autest]

**Pull request merged**